### ISSUE TITLE
[BUG] Import of cudf on Colab after install gives AttributeError around object `CudaRecordBatchStreamReader` 

### ISSUE THREAD

**Issue by taureandyernv** [rapidsai/cudf#3390]
**Describe the bug**
After successfully installing v0.11 nightlies into colab, when i `import cudf`, i get an AttributeError: `AttributeError: type object 'cudf._lib.gpuarrow.CudaRecordBatchStreamReader' has no attribute '__reduce_cython__',`.  Looking at the code, it was removed in `https://github.com/rapidsai/cudf/commit/e5192de83d7d3a568796cadfdd4159236d6ec665`.  

This does not happen if i use 0.10 stable 

**Steps/Code to reproduce bug**'
1. Open a new colab notebook.   
2. run the code below in a code cell
```
!wget -nc https://raw.githubusercontent.com/rapidsai/notebooks-contrib/taureandyernv-colab-p100/utils/rapids-colab.sh
!bash rapids-colab.sh

import sys, os

sys.path.append('/usr/local/lib/python3.6/site-packages/')
os.environ['NUMBAPRO_NVVM'] = '/usr/local/cuda/nvvm/lib64/libnvvm.so'
os.environ['NUMBAPRO_LIBDEVICE'] = '/usr/local/cuda/nvvm/libdevice/'
```
3. in a next code cell run `import cudf`

The packages installed in the rapids-colab.sh conda script:
```
conda install -y --prefix /usr/local \
      -c rapidsai-nightly/label/xgboost -c rapidsai-nightly -c nvidia -c conda-forge \
      python=3.6 cudatoolkit=10.1 \
      cudf=$RAPIDS_VERSION cuml cugraph gcsfs pynvml \
      dask-cudf \
      xgboost
```

**Expected behavior**
cudf successfully imports.  This happens if you run this on step 2 using the release repo:
```
!wget -nc https://github.com/rapidsai/notebooks-contrib/raw/master/utils/rapids-colab.sh
!bash rapids-colab.sh 0.10

import sys, os

sys.path.append('/usr/local/lib/python3.6/site-packages/')
os.environ['NUMBAPRO_NVVM'] = '/usr/local/cuda/nvvm/lib64/libnvvm.so'
os.environ['NUMBAPRO_LIBDEVICE'] = '/usr/local/cuda/nvvm/libdevice/'
```
**Environment overview (please complete the following information)**
 - Environment location: Google Colab
 - Method of cuDF install: conda, command is above

**Environment details**
`conda list | grep`

```
cudf                      0.11.0a191118         py36_2889    rapidsai-nightly
dask-cudf                 0.11.0a191118         py36_2889    rapidsai-nightly
libcudf                   0.11.0a191118     cuda10.1_2881    rapidsai-nightly

cugraph                   0.11.0a1191118         py36_121    rapidsai-nightly
libcugraph                0.11.0a1191118     cuda10.1_121    rapidsai-nightly
```

**Traceback**
```---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-22-9dd6c2bbcab3> in <module>()
----> 1 import cudf
      2 from collections import OrderedDict

4 frames
/usr/local/lib/python3.6/site-packages/cudf/__init__.py in <module>()
      2 
      3 
----> 4 from cudf import core, datasets
      5 from cudf._version import get_versions
      6 from cudf.core import DataFrame, Index, MultiIndex, Series, from_pandas, merge

/usr/local/lib/python3.6/site-packages/cudf/core/__init__.py in <module>()
      1 # Copyright (c) 2018-2019, NVIDIA CORPORATION.
----> 2 from cudf.core import buffer, column
      3 from cudf.core.buffer import Buffer
      4 from cudf.core.dataframe import DataFrame, from_pandas, merge
      5 from cudf.core.index import (

/usr/local/lib/python3.6/site-packages/cudf/core/column/__init__.py in <module>()
----> 1 from cudf.core.column.categorical import CategoricalColumn  # noqa: F401
      2 from cudf.core.column.column import (  # noqa: F401
      3     Column,
      4     TypedColumnBase,
      5     as_column,

/usr/local/lib/python3.6/site-packages/cudf/core/column/categorical.py in <module>()
      7 import pyarrow as pa
      8 
----> 9 import cudf._lib as libcudf
     10 from cudf.core.buffer import Buffer
     11 from cudf.core.column import column

/usr/local/lib/python3.6/site-packages/cudf/_lib/__init__.py in <module>()
----> 1 from . import (
      2     avro,
      3     binops,
      4     concat,
      5     copying,

cudf/_lib/gpuarrow.pyx in init cudf._lib.gpuarrow()

AttributeError: type object 'cudf._lib.gpuarrow.CudaRecordBatchStreamReader' has no attribute '__reduce_cython__'
```

**Additional context**
cugraph has a similar issue that also only happens in 0.11

**Comment by taureandyernv** [comment]
cc @randerzander

**Comment by ayushdg** [comment]
@taureandyernv can you do a 
```
! conda list | grep cudf
! conda list | grep cugraph
```
and share the outputs. I faced a different error recently but it was due to a mismatch between the libcudf and cudf version that Conda was picking up while installing the packages.

**Comment by taureandyernv** [comment]
@ayushdg I'll put it here and update the original post

```
cudf                      0.11.0a191118         py36_2889    rapidsai-nightly
dask-cudf                 0.11.0a191118         py36_2889    rapidsai-nightly
libcudf                   0.11.0a191118     cuda10.1_2881    rapidsai-nightly

cugraph                   0.11.0a1191118         py36_121    rapidsai-nightly
libcugraph                0.11.0a1191118     cuda10.1_121    rapidsai-nightly
```

**Comment by taureandyernv** [comment]
Changing script in original post as this was updated to work for the masses using 0.10 stable.

**Comment by kkraus14** [comment]
@taureandyernv looks like you have a version mismatch between libcudf and cudf there. Could you try again?

**Comment by taureandyernv** [comment]
@kkraus14 here is the output from this morning's attempt:
```
cudf                      0.11.0a191120         py36_3106    rapidsai-nightly
dask-cudf                 0.11.0a191120         py36_3106    rapidsai-nightly
libcudf                   0.11.0a191120     cuda10.1_3106    rapidsai-nightly
cugraph                   0.11.0a1191120         py36_121    rapidsai-nightly
libcugraph                0.11.0a1191120     cuda10.1_121    rapidsai-nightly
```

All versions are now the same, but i get the same errors.

**Comment by kkraus14** [comment]
To close out this issue, @taureandyernv and I explored for a while today and we found that Colab actually loads a bunch of modules before any code is executed but then deletes the top level variable that references the module. One of these modules happened to be pyarrow and version 0.14.1 is shipped with Colab. A workaround is incoming that will unload the 0.14.1 modules and then load the conda installed 0.15.0 modules correctly.

Closing this issue as nothing can be done from the cuDF side to alleviate this.

**Comment by ayushdg** [comment]
Update based on some experiments yesterday: I was able to reproduce this as well. Sometimes I got the error `PyArrow._lib` has no attribute `LargeListType` and sometimes I got the `__reduce_cython__` error. It might have to do with Colab environment weirdness and picking up the wrong versions of things? 
For reference we Conda install packages to `/usr/local` which makes it's way into the `site-packages` library folder. 
Colab also has certain libraries preinstalled in `dist-packages` such as pandas (0.25.2), PyArrow(0.14.1), Cython, numpy etc. I tried force installing PyArrow 0.15.0 in `dist-packages` and it got rid of the import errors. But I ended up breaking the environment in many ways and getting `CFFI` version mismatch errors when trying to create a column. (CFFI python was picking up the Cython `so` files from the wrong place).
